libs/core/src/lib/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,275 @@
+
+    constructor(private changeDetRef: ChangeDetectorRef) { }
+
+    /** @hidden */
+    onChange: Function = () => { };
+
+    /** @hidden */
+    onTouched: Function = () => { };
+
+
+
+    /** @hidden */
+    registerOnChange(fn: any): void {
+        this.onChange = fn;
+    }
+
+    /** @hidden */
+    registerOnTouched(fn: any): void {
+        this.onTouched = fn;
+    }
+
+    /** @hidden */
+    setDisabledState(isDisabled: boolean): void {
+        this.disabled = isDisabled;
+        this.changeDetRef.detectChanges();
+    }
+
+    /** @hidden */
+    writeValue(files: File[]): void {
+        // not needed - should be handled by user.
+    }
+
+    /** @hidden */
+    selectHandler(event: File[]) {
+
+        this.validateFiles(event);
+        if (this.valid_files.length > 0) {
+            this.setInputValue(this.valid_files);
+            this.onChange(this.valid_files);
+            this.selectedFilesChanged.emit(this.valid_files);
+        }
+        if (this.invalid_files.length > 0) {
+            this.selectedInvalidFiles.emit(this.invalid_files);
+        } else {
+            this.setInputValue(event);
+            this.onChange(event);
+            this.selectedFilesChanged.emit(event);
+        }
+
+    }
+
+    validateFiles(event: File[]) {
+        if (this.fileLimit && event.length > this.fileLimit) {
+            throw ('FileLimitError - Selected files count is more than specified limit ');
+        }
+
+        if (this.maxFileSize && this.minFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize && file.size > minsize) {
+                    this.valid_files.push(file);
+                } else {
+                    this.invalid_files.push(file);
+                }
+            });
+        } else if (this.maxFileSize) {
+            const maxsize = this.parseFileSize(this.maxFileSize);
+            event.forEach((file) => {
+                if (file.size < maxsize) {
+                    this.valid_files.push(file);
+                } else {
+                    this.invalid_files.push(file);
+                }
+            });
+        } else if (this.minFileSize) {
+            const minsize = this.parseFileSize(this.minFileSize);
+            event.forEach((file) => {
+                if (file.size > minsize) {
+                    this.valid_files.push(file);
+                } else {
+                    this.invalid_files.push(file);
+                }
+            });
+        }
+
+    }
+
+    setInputValue(selectedFiles: File[]) {
+        let fileName = '';
+        selectedFiles.forEach((file) => {
+            fileName = fileName.concat(' ' + file.name);
+        });
+        this.inputRefText.nativeElement.value = fileName;
+        this.inputRefText.nativeElement.title = fileName;
+    }
+
+    parseFileSize(filesize: string) {
```

Comment:
Can you add unit tests to the spec for this function which test all the valid inputs and a few invalid inputs as well?

Reply:
Written unit test cases for all the scenario.